src/main/scala/edu/ie3/powerFactory2psdm/model/PreprocessedPfGridModel.scala

```diff
@@ -198,14 +199,22 @@ object PreprocessedPfGridModel extends LazyLogging {
           )
         )
     }
-    val uniqueIds = modelIds.distinct
-    if (uniqueIds.size < modelIds.size) {
-      val duplicateIds = modelIds.diff(uniqueIds)
+    val duplicateIds = ConversionHelper.getDuplicates(modelIds.toVector)
```

Comment:
Unnecessary conversion

```suggestion
    val duplicateIds = ConversionHelper.getDuplicates(modelIds)
```